wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/convert/tool/kubernetes/KubernetesToolToWorkspaceApplier.java

```diff
@@ -190,47 +236,53 @@ private void estimateCommandsMachineName(
                         .equals(c.getAttributes().get(Constants.TOOL_NAME_COMMAND_ATTRIBUTE)))
             .collect(toList());
     if (toolCommands.isEmpty()) {
+      // tool has no any commands
       return;
     }
-    List<Pod> pods =
-        recipeObjects
-            .stream()
-            .filter(hasMetadata -> hasMetadata instanceof Pod)
-            .map(hasMetadata -> (Pod) hasMetadata)
-            .collect(toList());
 
-    Pod pod;
-    if (pods.size() != 1 || (pod = pods.get(0)).getSpec().getContainers().isEmpty()) {
-      // recipe contains several containers
+    Optional<Container> toolContainerOpt = findSingleContainer(toolPods);
+    if (!toolContainerOpt.isPresent()) {
+      // recipe contains no/several containers
```

Comment:
What alternatives do we have here? How is the command supposed to operate in this case?

Reply:
IDE will show an user dialog to choose a container manually